pandas/tests/frame/test_reductions.py

```diff
@@ -695,6 +710,42 @@ def test_sum_corner(self):
         assert len(axis0) == 0
         assert len(axis1) == 0
 
+    @pytest.mark.parametrize(
+        "index",
+        [
+            tm.makeRangeIndex(0),
+            tm.makeDateIndex(0),
+            tm.makeNumericIndex(0, dtype=int),
+            tm.makeNumericIndex(0, dtype=float),
+            tm.makeDateIndex(0, freq="M"),
+            tm.makePeriodIndex(0),
+        ],
+    )
+    def test_axis_1_empty(self, all_reductions, index, using_array_manager):
+        df = DataFrame(columns=["a"], index=index)
+        result = getattr(df, all_reductions)(axis=1)
+        expected_dtype = {
+            "any": "bool",
+            "all": "bool",
+            "count": "int64",
+            "sum": "float",
+            "prod": "float",
+            "skew": "float",
+            "kurt": "float",
+            "sem": "float",
+        }.get(all_reductions, "object")
+        if using_array_manager and all_reductions in (
+            "max",
+            "min",
+            "mean",
+            "std",
+            "var",
+            "median",
+        ):
+            expected_dtype = "float"
```

Comment:
On main, there is no behavior change for array manager in this PR. For block manager, the reducers on L738-743 return float type on main but object type here.